migrator/migrator-servlet/src/main/java/com/google/phonenumbers/ServletMain.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 The Libphonenumber Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.phonenumbers;
+
+import com.google.phonenumbers.migrator.MigrationFactory;
+import com.google.phonenumbers.migrator.MigrationJob;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(name = "Migrate", value = "/migrate")
+public class ServletMain extends HttpServlet {
+
+  /**
+   * Retrieves the form data for either a single number migration or a file migration from the index.jsp file and calls
+   * the relevant method to perform the migration
+   */
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String number = req.getParameter("number");
+    String countryCode;
+    if (number != null) {
+      countryCode = req.getParameter("numberCountryCode");
```

Comment:
Is it possible for countryCode to be null even though number is set?

Reply:
The countryCode here will always be set if number isn't null because the form in the jsp file has it as a required field. So the user won't be able to hit migrate until they have filled it out.

Reply:
Sounds good!

Reply:
I believe we shouldn't rely on front-end validation in the back-end - malicious users can easily change the properties of the field to be not required anymore in the browser's DOM. As such, I recommend validating the field here by performing a nullness check.

Reply:
validation of country code now done below